src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,151 @@
+                "Illegal interval: Attempting to raise OutOfBoundsError, but value is not out of bounds.",
+            )
+        super().__init__(f"{actual} is not inside {lower_bound.str_lower_bound()}, {upper_bound.str_upper_bound()}.")
+
+
+class Bound(ABC):
+    """Abstract base class for (lower or upper) Bounds on a float value."""
+
+    def __init__(self, value: float):
+        self._value = value
+
+    def __str__(self) -> str:
+        """Get a string representation of the concrete value of the Bound."""
+        return str(self._value)
+
+    @abstractmethod
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the Bound as the lower Bound of an interval."""
+
+    @abstractmethod
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the Bound as the upper Bound of an interval."""
+
+    @abstractmethod
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value does not exceed the Bound on the lower side."""
+
+    @abstractmethod
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value does not exceed the Bound on the upper side."""
+
+
+class ClosedBound(Bound):
+    """A closed Bound, i.e. the value on the border belongs to the range."""
+
+    def __init__(self, value: float):
+        super().__init__(value)
+
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the lower Bound of an interval."""
+        return f"[{self}"
+
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the ClosedBound as the upper Bound of an interval."""
+        return f"{self}]"
+
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value is not strictly lower than the ClosedBound."""
+        return value >= self._value
+
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value is not strictly higher than the ClosedBound."""
+        return value <= self._value
+
+
+class OpenBound(Bound):
+    """
+    An open Bound, i.e. the value on the border does not belong to the range.
+
+    May be infinite (unbounded).
+    """
+
+    def __init__(self, value: float):
+        super().__init__(value)
+
+    def str_lower_bound(self) -> str:
+        """Get a string representation of the OpenBound as the lower Bound of an interval."""
+        return f"({self}"
+
+    def str_upper_bound(self) -> str:
+        """Get a string representation of the OpenBound as the upper Bound of an interval."""
+        return f"{self})"
+
+    def check_lower_bound(self, value: float) -> bool:
+        """Check that a value is not lower or equal to the Bound."""
+        return value > self._value
+
+    def check_upper_bound(self, value: float) -> bool:
+        """Check that a value is not higher ot equal to the Bound."""
+        return value < self._value
+
+
+class Infinity(OpenBound):
```

Comment:
I don't see the advantage of having `Infinity` and `MinInfinity` classes over only encoding a missing lower/upper bound as `None`. Now there are two ways to express the same thing.

Reply:
Yes, it was mainly to have polymorphism for the stringification.
Made it private and removed it from the tests.

However, the tests also need to test the `check_lower_bound` and `check_upper_bound` methods, so the `_Infinity` and `_MinInfinity` classes need to remain in `__init.py__` in order to stay importable as far as I understand?

Reply:
No, you can also directly import declarations from the module they are defined in. The `__init__.py` files are just use to gather declarations from several files.

Reply:
Done.